parts/plugins/x_builddeb.py

```diff
@@ -30,13 +30,18 @@ def patch_snapcraft():
 class XBuildDeb(snapcraft.BasePlugin):
 
     def build(self):
+        super().build()
         self.run(["sudo", "apt-get", "build-dep", "-y", "./"])
         # XXX: get this from "debian/gbp.conf:postexport"
         self.run(["./get-deps.sh"])
-        # run the real build, -ptrue means run "true" to sign the package.
-        # Newer dpkg-buildpackage has "--no-sign" but not the xenial version
-        self.run(["dpkg-buildpackage", "-ptrue"])
+        env=os.environ
+        if os.getuid() == 0:
+            # disable running the tests during the build when run as root
+            # because quite a few of them will break
+            env["DEB_BUILD_OPTIONS"] = "nocheck"
```

Comment:
You're directly modifying `os.environ` here, is that intended? If not, you may want to use `os.environ.copy()` above.

Reply:
Good point, thank you!